data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -230,4 +247,10 @@ private SourcePartitionStoreItem validateAndGetSourcePartitionStoreItem(final St
 
         return optionalPartitionItem.get();
     }
+
+    private void validateIsInitialized() {
+        if (!initialized) {
```

Comment:
Is there any alternate approach we can take here? Like calling initialize() here instead of throwing?

Reply:
We could, but I think this is a fair exception to throw. It is preferred to have an explicit call